server/src/frontend_api.rs

```diff
@@ -106,23 +97,24 @@ async fn post_proxy_all_features(
 #[post("/frontend/all")]
 async fn post_frontend_all_features(
     edge_token: EdgeToken,
-    features_source: web::Data<dyn EdgeSource>,
+    engine_cache: web::Data<DashMap<String, EngineState>>,
     context: web::Json<Context>,
 ) -> EdgeJsonResult<FrontendResult> {
-    post_all_features(edge_token, features_source, context).await
+    post_all_features(edge_token, engine_cache, context)
 }
 
-async fn post_all_features(
+fn post_all_features(
     edge_token: EdgeToken,
-    features_source: web::Data<dyn EdgeSource>,
+    engine_cache: web::Data<DashMap<String, EngineState>>,
     context: web::Json<Context>,
 ) -> EdgeJsonResult<FrontendResult> {
-    let client_features = features_source.get_client_features(&edge_token).await;
     let context = context.into_inner();
-
-    let toggles = resolve_frontend_features(client_features?, context).collect();
-
-    Ok(Json(FrontendResult { toggles }))
+    let engine = edge_token
+        .environment
+        .and_then(|k| engine_cache.get(&k))
+        .ok_or_else(|| EdgeError::PersistenceError("Could not find data for token".into()))?;
```

Comment:
It's very uncertain. This flow can happen for one of two reasons: your data isn't there right now but it's coming in the near future or your data is never going to be there because this token doesn't match any data that's being refreshed. It's impossible to tell which is which from Edge's perspective. I think we want to rework our failure cases for a v1.1